Applications/Utils/ModelPreparation/PartitionMesh/MeshPartitioning.cpp

```diff
@@ -0,0 +1,493 @@
+
+    for (std::size_t i=0; i<nnodes; i++)
+    {
+        std::size_t part_id;
+        npart_in >> part_id >> std::ws;
+        nodes_partition_ids.push_back(part_id);
+    }
+    npart_in.close();
+    //TEST  std::remove(fname_parts.c_str());
+
+    // -- Open files for output
+    //
+    // for ASCII output
+    std::fstream os_subd;
+    std::fstream os_subd_head;
+    std::fstream os_subd_node;
+    // for binary output
+    FILE *of_bin_cfg = 0;   // File to output headers
+    FILE *of_bin_nod = 0;   // File to output an array of all nodes
+    FILE *of_bin_ele = 0;   // File to output an array of all non-ghost elements
+    FILE *of_bin_ele_g = 0; // File to output an array of all ghost elements
+
+    if (output_binary)
+    {
+        std::string fname = file_name_base + "_partitioned_msh_cfg" + npartitions_str + ".bin";
+        of_bin_cfg = fopen(fname.c_str(), "wb");
+
+        fname = file_name_base + "_partitioned_msh_ele" + npartitions_str + ".bin";
+        of_bin_ele = fopen(fname.c_str(), "wb");
+
+        fname = file_name_base + "_partitioned_msh_ele_g" + npartitions_str + ".bin";
+        of_bin_ele_g = fopen(fname.c_str(), "wb");
+
+        fname = file_name_base + "_partitioned_msh_nod" + npartitions_str + ".bin";
+        of_bin_nod = fopen(fname.c_str(), "wb");
+    }
+    else
+    {
+        std::string fname = file_name_base + "_partitioned_cfg" + npartitions_str + ".msh";
+        os_subd_head.open(fname.c_str(), std::ios::out|std::ios::trunc );
+        const std::string mesh_info = "Subdomain mesh "
+                                      "(Number of non-ghost nodes;  Number of non ghost base nodes; Number of elements; "
+                                      "Number of ghost elements; Number of base nodes; Number of nodes) "
+                                      "Number of the base nodes of the global mesh; Number of the nodes of the global mesh; "
+                                      "Number of integers to define elements; Number of integers to define ghost elements; "
+                                      "Reserved number.";
+        os_subd_head << mesh_info << std::endl;
+        os_subd_head << npartitions << std::endl;
+
+        fname = file_name_base + "_partitioned_elems_" + npartitions_str + ".msh";
+        os_subd.open(fname.c_str(), std::ios::out|std::ios::trunc );
+
+        fname = file_name_base + "_partitioned_nodes_" + npartitions_str + ".msh";
+        os_subd_node.open(fname.c_str(), std::ios::out|std::ios::trunc );
+
+        std::setw(14);
+        os_subd.precision(14);
+        os_subd.setf(std::ios::scientific);
+    }
+
+    PetscInt global_node_id_offset = 0;
+    std::vector<bool> nodes_part_reserved;
+    nodes_part_reserved.reserve(nnodes);
+    std::vector<PetscInt> node_global_ids;
+    node_global_ids.reserve(nnodes);
+    for (std::size_t i=0; i<nnodes; i++)
+    {
+        nodes_part_reserved.push_back(false);
+        node_global_ids.push_back(0);
```

Comment:
Both vectors can be directly created with correct size and a given value.
See http://en.cppreference.com/w/cpp/container/vector/vector (2)